ui/apps/platform/src/hooks/useFeatureFlags.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { createContext, useContext, ReactNode, useMemo } from 'react';
+
+import type { FeatureFlagEnvVar } from 'types/featureFlag';
+import { fetchFeatureFlags } from 'services/FeatureFlagsService';
+import useRestQuery from './useRestQuery';
+
+export type IsFeatureFlagEnabled = (envVar: FeatureFlagEnvVar) => boolean;
+
+type FeatureFlagsContextType = {
+    isLoadingFeatureFlags: boolean;
+    error: Error | null;
+    isFeatureFlagEnabled: IsFeatureFlagEnabled;
+};
+
+const FeatureFlagsContext = createContext<FeatureFlagsContextType | undefined>(undefined);
+
+type FeatureFlagsProviderProps = {
+    children: ReactNode;
+};
+
+export function FeatureFlagsProvider({ children }: FeatureFlagsProviderProps) {
+    const { data, isLoading: isLoadingFeatureFlags, error } = useRestQuery(fetchFeatureFlags);
+
+    const value: FeatureFlagsContextType = useMemo(
+        () => ({
+            isLoadingFeatureFlags,
+            error: error || null,
```

Comment:
Forgive me for not searching deeper to answer question: `||` for null string also or `??` for nullish values only?